testflinger_agent/client.py

```diff
@@ -64,21 +64,21 @@ def _configure_influx(self):
 
         :return: influxdb object or None
         """
-        user = os.environ.get("INFLUX_USER", "")
-        password = os.environ.get("INFLUX_PW", "")
-        port = int(os.environ.get("INFLUX_PORT", 8086))
         host = os.environ.get("INFLUX_HOST")
         if not host:
             logger.error("InfluxDB host undefined")
-            return
+            return None
+        port = int(os.environ.get("INFLUX_PORT", 8086))
+        user = os.environ.get("INFLUX_USER", "")
+        password = os.environ.get("INFLUX_PW", "")
 
         influx_client = InfluxDBClient(
             host, port, user, password, self.influx_agent_db
         )
 
-        # ensure we can connect to influxdb host
+        # ensure we can connect to influxdb
         try:
-            influx_client.ping()
+            influx_client.create_database(self.influx_agent_db)
         except ConnectionError as exc:
```

Comment:
I think I see a problem here. If create_database() fails, then it will log an error, that's good. But then it will still return influx_client, which we think is actually not working. Then if we try to use it later, it's going to fail.
I think the original intent was to return None in that case, correct?
another, possibly better approach would be to have it set an instance variable for an influx client so that when you try to send influx metrics, it either uses a good, working client -or- it sees that a working client hasn't been setup and just returns.  Does that make sense?

Either way, I think we need to handle this potential problem where you could get a non-working client back.

Reply:
Ok, I see that there's not an explicit return, but there's an else clause where it would only return the influx client if the create_database() happened. I still think something more self-contained like I described earlier would be nicer to have here, so Adrian is going to add a card to address this. I believe that this could be easily converted to something like that later with minimal changes to the source.